tests/e2e/vm_label_annotation_test.go

```diff
@@ -18,263 +18,295 @@ package e2e
+					vmPodObj := v1.Pod{}
+					err = GetObject(kc.ResourcePod, activePod, &vmPodObj, kc.GetOptions{Namespace: conf.Namespace})
+					Expect(err).NotTo(HaveOccurred(), err)
+					Expect(vmPodObj.Labels).Should(HaveKeyWithValue("specialKey", "specialValue"))
+				}
 			})
-			It("POD virtlauncher", func() {
-				pod := getPodName(vm.Name)
-				res := getRecourseLabel(kc.ResourcePod, pod)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get pod %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
+
+			It(fmt.Sprintf("removes label %s from VMs", specialKeyValue), func() {
+				res := kubectl.List(kc.ResourceVM, kc.GetOptions{
+					Labels:    testCaseLabel,
+					Namespace: conf.Namespace,
+					Output:    "jsonpath='{.items[*].metadata.name}'",
+				})
+				Expect(res.Error()).NotTo(HaveOccurred(), "cmd: %s\nstderr: %s", res.GetCmd(), res.StdErr())
+
+				vms := strings.Split(res.StdOut(), " ")
+				err := RemoveLabel(kc.ResourceVM, specialKeyValue, vms...)
+				Expect(err).NotTo(HaveOccurred(), err)
 			})
-		})
-	})
 
-	Context("Annotation", func() {
-		var vm *virtv2.VirtualMachine
+			It("checks VMs and pods labels after VMs unlabeling", func() {
+				res := kubectl.List(kc.ResourceVM, kc.GetOptions{
+					Labels:    testCaseLabel,
+					Namespace: conf.Namespace,
+					Output:    "jsonpath='{.items[*].metadata.name}'",
+				})
+				Expect(res.Error()).NotTo(HaveOccurred(), "cmd: %s\nstderr: %s", res.GetCmd(), res.StdErr())
 
-		BeforeAll(func() {
-			By("Apply manifest")
-			var err error
-			vm, err = GetVMFromManifest(vmManifest)
-			Expect(err).NotTo(HaveOccurred())
-			ApplyFromFile(vmManifest)
-			WaitVmStatus(vm.Name, VMStatusRunning)
-		})
+				vms := strings.Split(res.StdOut(), " ")
+				for _, vm := range vms {
+					vmObj := virtv2.VirtualMachine{}
+					err := GetObject(kc.ResourceVM, vm, &vmObj, kc.GetOptions{Namespace: conf.Namespace})
+					Expect(err).NotTo(HaveOccurred(), err)
+					Expect(vmObj.Labels).ShouldNot(HaveKey("specialKey"))
 
-		AfterAll(func() {
-			By("Delete manifest")
-			kubectl.Delete(kc.DeleteOptions{
-				Filename:       []string{vmManifest},
-				FilenameOption: kc.Filename,
+					activePod := GetActiveVirtualMachinePod(&vmObj)
+					vmPodObj := v1.Pod{}
+					err = GetObject(kc.ResourcePod, activePod, &vmPodObj, kc.GetOptions{Namespace: conf.Namespace})
+					Expect(err).NotTo(HaveOccurred(), err)
+					Expect(vmPodObj.Labels).ShouldNot(HaveKey("specialKey"))
+				}
 			})
 		})
+	})
 
-		Describe(fmt.Sprintf("Add annotation %s=%s", annotationName, annotationValue), func() {
-			It("Annotated", func() {
-				subCMD := fmt.Sprintf("-n %s annotate vm %s %s=%s", conf.Namespace, vm.Name, annotationName,
-					annotationValue)
-				res := kubectl.RawCommand(subCMD, ShortWaitDuration)
-				Expect(res.Error()).NotTo(HaveOccurred())
-			})
-		})
+	Describe("Annotating", func() {
+		Context(fmt.Sprintf("When virtual machines in %s phase:", PhaseRunning), func() {
```

Comment:
There is only one Context in this Describe. It seems more convenient to merge them into single Context:
```
Context(fmt.Sprintf("Annotate VirtualMachines in %s phase", PhaseRunning), func() ...
```